### ISSUE TITLE
Query: Compiler crash for invalid Include path

### ISSUE THREAD

**Issue by jnm2** [dotnet/efcore#7516]
([As asked](https://github.com/aspnet/EntityFramework/issues/7424#issuecomment-276194553) by @anpete.)

Please support eager loading when the root entity ends up in a projection.

There is not a good way to do what I need to do: load a list of entities with multiple child collections on each entity populated, plus an arbitrary calculation associated to each entity.

This would be perfect:
```c#
var results = dbContext.Entities
    .Select(entity => new
    {
        Entity = entity,
        ComplicatedServerSideCalculation = Entity.NavProperty.OtherInfo.Sum(x => x * 2)
    })
    .Include(info => info.Entity.ChildCollection1)
    .Include(info => info.Entity.ChildCollection2)
    .Include(info => info.Entity.ChildCollection3)
    .ToListAsync();
```

Intuitively, I'd also expect that this would work:
```c#
var results = dbContext.Entities
    .Include(entity => entity.ChildCollection1)
    .Include(entity => entity.ChildCollection2)
    .Include(entity => entity.ChildCollection3)
    .Select(entity => new
    {
        Entity = entity,
        ComplicatedServerSideCalculation = Entity.NavProperty.OtherInfo.Sum(x => x * 2)
    })
    .ToListAsync();
```

**Comment by rowanmiller** [comment]
Have you checked if the second code listing works or not? We think that will already work.

Also, what happens with the first example (exception, ignored, etc.)?

**Comment by jnm2** [comment]
@rowanmiller You're right about the second code sample! I'm surprised and happy! Given the response to the [exact same question for EF6](https://entityframework.codeplex.com/workitem/2939) I thought this was a path you decided against. This will be a major win for us when we finally get to switch to EF Core.

As for the first code sample with EF Core, I get:

```
System.InvalidCastException occurred
  HResult=0x80004002
  Message=Unable to cast object of type 'System.Linq.Expressions.NewExpression' to type 'System.Linq.Expressions.MemberExpression'.
  Source=Microsoft.EntityFrameworkCore
  StackTrace:
   at Microsoft.EntityFrameworkCore.Query.ResultOperators.Internal.IncludeResultOperator.GetQuerySource(MemberExpression expression)
   at Microsoft.EntityFrameworkCore.Query.ResultOperators.Internal.IncludeResultOperator.GetQuerySource(MemberExpression expression)
   at Microsoft.EntityFrameworkCore.Query.ResultOperators.Internal.IncludeExpressionNode.CreateResultOperator(ClauseGenerationContext clauseGenerationContext)
   at Remotion.Linq.Parsing.Structure.IntermediateModel.ResultOperatorExpressionNodeBase.ApplyNodeSpecificSemantics(QueryModel queryModel, ClauseGenerationContext clauseGenerationContext)
   at Remotion.Linq.Parsing.Structure.IntermediateModel.MethodCallExpressionNodeBase.Apply(QueryModel queryModel, ClauseGenerationContext clauseGenerationContext)
   at Remotion.Linq.Parsing.Structure.QueryParser.GetParsedQuery(Expression expressionTreeRoot)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.CompileQueryCore[TResult](Expression query, INodeTypeProvider nodeTypeProvider, IDatabase database, ILogger logger, Type contextType)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.<>c__DisplayClass19_0`1.<CompileQuery>b__0()
   at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQueryCore[TFunc](Object cacheKey, Func`1 compiler)
   at Microsoft.EntityFrameworkCore.Query.Internal.CompiledQueryCache.GetOrAddQuery[TResult](Object cacheKey, Func`1 compiler)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.CompileQuery[TResult](Expression query)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryCompiler.Execute[TResult](Expression query)
   at Microsoft.EntityFrameworkCore.Query.Internal.EntityQueryProvider.Execute[TResult](Expression expression)
   at Remotion.Linq.QueryableBase`1.GetEnumerator()
   at Microsoft.EntityFrameworkCore.EntityFrameworkQueryableExtensions.IncludableQueryable`2.GetEnumerator()
   at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
   at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
   at Test_EF_eager_loading.Program.Main() in c:\users\jmusser\documents\visual studio 2017\Projects\Test EF eager loading\Test EF eager loading\Program.cs:line 46
```

**Comment by maumar** [comment]
We decided not to support the first scenario. Include only works on entities.

**Comment by smitpatel** [comment]
Instead of throwing from compiler crash, we should print out exception message that include is not applied on entity type.

**Comment by anpete** [comment]
We already warn for ignored Includes. Agree we should never crash though.

**Comment by maumar** [comment]
Fixed dbe6b2846bc81e2653605eb4c2b6b4a7987848e5 in a sense that we no longer throw for this case and log Ignored Include warning instead

**Comment by maumar** [comment]
Reactivating and sending for re-triage - we might end up supporting this scenario in the future

**Comment by maumar** [comment]
Modified the original fix in 2e4e138572404ed936eb6e75cde4ed40a8ba6096. Now we throw NotSupportedException with a better explanation, rather than just log a warning.
